### ISSUE TITLE
Floodgate Bot

### ISSUE THREAD

**Issue by js-choi** [tc39/proposal-pipeline-operator#231]
Heya hey, everyone. I’m a member of the TC39 pipe champion group, and I’m also a volunteer moderator here.
I wanted to make an announcement. (It shouldn’t actually affect things much.)
We’ve installed a [“floodgate” moderation bot](https://github.com/marketplace/actions/comment-floodgate).
When a issue gets a huge flood of new comments (i.e., new comments are being created at a very high rate),
then the bot will temporarily lock that issue.
(The threshold rate is currently set to twelve messages in the same thread in the same sixty minutes.)

This doesn’t mean that the automatically locked high-traffic threads will get locked permanently.
It just means that the issue’s thread will cool down until a volunteer moderator is able to look at what’s going on.

Why are we trying this out?
Well, we’ve had two cases so far in which “[centithreads](https://www.urbandictionary.com/define.php?term=centithread)” suddenly occur in the course of hours.
These cases were very difficult for almost everyone (myself included) to keep up with,
and much of their contents didn’t meet the minimum standards of the [TC39 Code of Conduct](https://tc39.es/code-of-conduct/).

The members of the champion team are all just unpaid volunteers, with our own families and our own separate day jobs.

We do our best to read every message and piece of feedback that comes through, but
when a hundred (often lengthy) messages occur every day, it’s not possible for us to keep up.

The bot just would automatically force a cool down
until a volunteer moderator can come look at stuff and ensure everything is okay,
possibly with a warning or some message-hiding, can keep tempers and conduct from spiraling out of control.
If the thread is actually okay, the volunteer moderate can unlock it.
And in the meantime people can take a break from the high-traffic issue.

We understand that there’s a risk of this accidentally triggering
when people are having rapid but good-faith conversations.
If this happens, we probably would adjust the bot’s trigger threshold.
But having a temporary cooldown after an hour of one new message every five minutes
probably wouldn’t be such a bad thing even then either.

We’re open to feedback about this.
TL;DR: Basically, we just want to avoid a centithread spawning overnight or over the weekend.
The bot will cause a temporary cooldowns, and the cooldown doesn’t mean permanent locking of a thread.
Hopefully this all makes sense. :)
